File: Anya_Seth/Sliding Window/Maximum Number of Vowels in a Substring of Given Length.java
class Solution {
    public int maxVowels(String s, int k) {
       int start = 0;
       int end = 0;
       int max = 0;
       int count = 0;
       while(end < s.length()){
        k--;
        if(s.charAt(end) == 'a' || s.charAt(end) == 'e' ||s.charAt(end) == 'i' || 
        s.charAt(end) == 'o' || s.charAt(end) == 'u'){
            count++;
        }
        while(k < 0 && start < s.length() ){
            if(s.charAt(start) == 'a' || s.charAt(start) == 'e' ||s.charAt(start) == 'i' || 
              s.charAt(start) == 'o' || s.charAt(start) == 'u') {
                count--;
            }
             k++;
            start++;
        }
        
        max = Math.max(max, count);
        
        end++;
       }
       return max;
    }
}

File: NikhilKumarGupta/Slidingwindow/Maximum Number of Vowels in a Substring of Given Length.cpp
class Solution {
    public:
    bool vowel(char c) {
        return (c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u');
    }
        int maxVowels(string s, int k) {
            int n=s.size();
            int start=0;
            int maxcount=0;
            int count =0;
    
            for(int end=0;end<n;end++){
                if(vowel(s[end])){
                    count++;
                   
                }
                if(end>=k-1){
                    maxcount=max(count,maxcount);
                
                if(vowel(s[start])){
                    count--;
                }
                start++;
                
    
            }
            }
            return maxcount;
            
          
            
        }
    };